sql/partition_info.h

```diff
@@ -67,7 +67,25 @@ struct Vers_part_info : public Sql_alloc
     }
     return false;
   }
-  my_time_t interval;
+  interval_type get_interval_type() const
+  {
+    DBUG_ASSERT(!interval.second_part);
+    DBUG_ASSERT(!interval.neg);
+
+    if (interval.year)
+      return INTERVAL_YEAR;
+    if (interval.month)
+      return INTERVAL_MONTH;
+    if (interval.day)
+      return INTERVAL_DAY;
+    if (interval.hour)
+      return INTERVAL_HOUR;
+    if (interval.minute)
+      return INTERVAL_MINUTE;
+    DBUG_ASSERT(interval.second);
+    return INTERVAL_SECOND;
+  }
+  INTERVAL interval;
```

Comment:
1. Store `ulong` instead of INTERVAL. Store `interval_type`, don't make ugly forks of type detection (or value detection).